coresimd/ppvt/api/fmt.rs

```diff
@@ -0,0 +1,53 @@
+//! Implements formating traits.
+
+macro_rules! impl_hex_fmt {
+    ($id:ident, $elem_ty:ident) => {
+        impl fmt::LowerHex for $id {
+            fn fmt(&self, f: &mut fmt::Formatter)
+                   -> fmt::Result {
+                write!(f, "{}(", stringify!($id))?;
+                let n = mem::size_of_val(self)
+                    / mem::size_of::<$elem_ty>();
+                for i in 0..n {
+                    if i > 0 {
+                        write!(f, ", ")?;
+                    }
+                    write!(f, "{:#x}", self.extract(i))?;
```

Comment:
I think here if you do `self.extract(i).fmt(f)` it'll automatically forward formatting flags like `#` which means we may not need to hardcode the `#`

Reply:
Done.